wave_lang/kernel/wave/scheduling/multi_buffering.py

```diff
@@ -24,9 +24,36 @@
     get_custom,
 )
 from ..utils.mapping_utils import get_dict_with_updated_key
+from .schedule_enums import SchedulingType
 
 
-def multi_buffer(trace: CapturedTrace):
+def _increase_dimensions_for_multibuffering(
+    original_buffer, reduction_axis, buffer_count
+):
+    # Create new shape with increased non-reduction dimensions
+    reduction_dim_indices = [
+        i for i, dim in enumerate(original_buffer.shape) if dim == reduction_axis
+    ]
+    new_shape = []
+    new_distributed_shape = []
+
+    for i, dim in enumerate(original_buffer.shape):
+        if i in reduction_dim_indices:
+            # Keep reduction dimensions as is
+            new_shape.append(dim)
+            new_distributed_shape.append(original_buffer.distributed_shape[i])
+        else:
+            # Increase non-reduction dimensions
+            new_shape.append(dim * buffer_count)
+            new_distributed_shape.append(
+                original_buffer.distributed_shape[i] * buffer_count
+            )
+
+    original_buffer.update_arg(0, tuple(new_shape))
+    original_buffer.update_arg(1, tuple(new_distributed_shape))
```

Comment:
I think this is fine for this PR, but in future PR, maybe would be good to have the expanded dimension the slowest/first dimension instead of the fastest/last dimension. CC: @harsh-nod

Reply:
Yes, I would agree that it should be the slowest dimension.